core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,663 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for classification algorithms."""
+
+import collections
+import keyword
+import StringIO
+import token
+import tokenize
+
+from core.classifiers import base
+
+import numpy as np
+from sklearn.feature_extraction import text as sklearn_text
+from sklearn import model_selection
+from sklearn import svm
+
+# Token which will be used to replace variable and method name when tokenizing
+# program.
+VAR_TOKEN = 'V'
+
+# Token which will be used to replace unkownn token when tokenizing
+# program.
+UNK_TOKEN = 'UNK'
+
+# T is minimum length of a substring that winnowing should match when
+# comparing two programs.
+# Minimum value of T. T is model parameter of winnowing algorithm.
+T_MIN = 3
+# Maximum value of T. T is model parameter of winnowing algorithm.
+T_MAX = 11
+
+# K is length of substring for which hash value is to be generated. Each
+# substring of length K in program will be converted to equivalent hash
+# value (K <= T).
+# Minimum value of K (K <= T). K is model parameter of winnowing algorithm.
+K_MIN = 3
+
+def get_token(program):
+    """Generate tokens for program using tokenize module.
+
+    Args:
+        program: str. Input program for which tokens are to be generated.
+
+    Yields:
+        int. Integer value which represents type of token.
+        str. Token.
+    """
+    for token_id, token_name, _, _, _ in tokenize.generate_tokens(
+            StringIO.StringIO(program).readline):
+        yield (token_id, token_name)
+
+
+def cv_tokenizer(program, var_token=VAR_TOKEN):
```

Comment:
OK. They are not required to be args, we can hard code them in function directly but I thought may be if we want to change it somewhere. I don't think we need that as args anyway.